extensions/terminal-suggest/src/shell/zsh.ts

```diff
@@ -65,67 +67,62 @@ async function getBuiltins(
 
 async function createCommandDescriptionsCache(options: ExecOptionsWithStringEncoding): Promise<void> {
 	cachedCommandDescriptions = new Map();
+	let output = '';
 	try {
-		let output = (await execHelper('man zshbuiltins', options));
-		if (output) {
-			// Extract bold words
-			const boldRegex = /(?:\w\x08\w)+/g;
-			const boldMatches = output.match(boldRegex) || [];
-			const boldCommands = new Set<string>();
-
-			// Process each bold match and clean it up (remove backspaces)
-			for (const match of boldMatches) {
-				const cleaned = match.replace(/\x08./g, '');
-				boldCommands.add(cleaned);
-			}
-
-			// Strip all backspaces from the output
-			output = output.replace(/.\x08/g, '');
-			const lines = output.split('\n');
-
-			let command: string | undefined;
-			let description: string[] = [];
+		const filePath = path.join(__dirname, '../../zshbuiltins.json');
+		const fileData = await readFile(filePath, 'utf8');
+		const json = JSON.parse(fileData);
+		output = json.output;
+	} catch {
+		// Fallback: run man command if JSON file is not available
+		output = await execHelper('man zshbuiltins', options);
+	}
```

Comment:
What I'm thinking this should look like eventually is to generate actual fig specs for these and then copy them over to something like `<userdatadir>/terminal/completions/zsh-builtins.ts` when we support custom fig specs (https://github.com/microsoft/vscode/issues/240240#issuecomment-2665967925).

We can avoid this parsing and just do that in pullZshBuiltins.ts instead in the meantime to simplify things and improve runtime perf.